src/box/alter.cc

```diff
@@ -264,6 +265,44 @@ index_opts_decode(struct index_opts *opts, const char *map,
 	return 0;
 }
 
+/** Decode an optional hostname field from the tuple. */
+static int
+tuple_field_tt_hostname(struct tt_hostname *out, struct tuple *tuple,
+			uint32_t fieldno, const char *field_name)
+{
+	uint32_t len = 0;
+	const char *field, *str = NULL;
+	if (tuple == NULL)
+		goto success;
+	field = tuple_field(tuple, fieldno);
+	if (field == NULL || mp_typeof(*field) == MP_NIL)
+		goto success;
+	str = mp_decode_str(&field, &len);
```

Comment:
Please don't forget to check that field is actually a MP_STR:
```
tarantool> box.space._schema:select{}
---
- - ['cluster_name', 'cluster']
  - ['replicaset_uuid', '48f8b244-b8ac-4866-9c67-54ac76433971']
  - ['version', 3, 0, 0]
...

tarantool> box.space._schema:replace{'cluster_name', 123}
Assertion failed: (0), function mp_decode_strl, file msgpuck.h, line 2544.
```

Reply:
Thanks for noticing, fixed and added tests for all 3 names.